src/test/java/entities/ExpenseFactoryTest.java

```diff
@@ -0,0 +1,102 @@
+
+        Expense expense = (Expense) monthObject;
+        Assertions.assertEquals(expense.getName(), inputData.getName());
+        Assertions.assertEquals(expense.getCategory(), inputData.getCategory());
+        Assertions.assertEquals(expense.getValue(), inputData.getValue());
+        Assertions.assertEquals(expense, inputData.getExpense());
+    }
+
+    /**
+     * Tests editMonthObject() with valid expense information and a different category.
+     */
+    @Test
+    public void ExpenseEditCategory() {
+        Category c1 = new Category("Dine out", 50);
+        Category c2 = new Category("Groceries", 100);
+        Expense e1 = new Expense("Subway", c1, 10);
+        ExpenseEditorInputData inputData = new ExpenseEditorInputData("Subway", c2, 10, e1);
+        MonthObjectFactory editor = new ExpenseFactory();
+        MonthObject monthObject = editor.editMonthObject(inputData);
+
+        Expense expense = (Expense) monthObject;
+        Assertions.assertEquals(expense.getName(), inputData.getName());
+        Assertions.assertEquals(expense.getCategory(), inputData.getCategory());
+        Assertions.assertEquals(expense.getValue(), inputData.getValue());
+        Assertions.assertEquals(expense, inputData.getExpense());
+    }
+
+    /**
+     * Tests editMonthObject() with valid expense information and a different category.
+     */
+    @Test
+    public void ExpenseEditValue() {
+        Category c1 = new Category("Dine out", 50);
+        Expense e1 = new Expense("Subway", c1, 10);
+        ExpenseEditorInputData inputData = new ExpenseEditorInputData("Subway", c1, 20, e1);
+        MonthObjectFactory editor = new ExpenseFactory();
+        MonthObject monthObject = editor.editMonthObject(inputData);
+
+        Expense expense = (Expense) monthObject;
+        Assertions.assertEquals(expense.getName(), inputData.getName());
+        Assertions.assertEquals(expense.getCategory(), inputData.getCategory());
+        Assertions.assertEquals(expense.getValue(), inputData.getValue());
+        Assertions.assertEquals(expense, inputData.getExpense());
+    }
+
+    /**
+     * Tests editMonthObject() with valid expense information and nothing different.
+     */
+    @Test
+    public void ExpenseEditNothing() {
+        Category c1 = new Category("Dine out", 50);
+        Expense e1 = new Expense("Subway", c1, 10);
+        ExpenseEditorInputData inputData = new ExpenseEditorInputData("Subway", c1, 10, e1);
+        MonthObjectFactory editor = new ExpenseFactory();
+        MonthObject monthObject = editor.editMonthObject(inputData);
+
+        Expense expense = (Expense) monthObject;
+        Assertions.assertEquals(expense.getName(), inputData.getName());
+        Assertions.assertEquals(expense.getCategory(), inputData.getCategory());
+        Assertions.assertEquals(expense.getValue(), inputData.getValue());
+        Assertions.assertEquals(expense, inputData.getExpense());
+    }
```

Comment:
Is this test needed? Is so, wouldn't we also need a similar test for Category editing?

Reply:
I would say we can get rid of this one since there is no related fail messages returned in add/edit features. Such that when there are no modifications, the UCI handles it as a success edit!

Reply:
I added it because of what Lulu is saying here, since we allow the user to edit nothing it is considered a success case and thus would be a possible input for the ExpenseFactory. It is not a useful test per say but I did want to include one test case for every possible option the user could do. If you feel it is unnecessary I can remove it and if you feel it is okay to stay I will add another one in the CategoryFactoryTest!

Reply:
Oh ok that makes sense. It is technically a bit redundant, but it is okay since it is explicitly handling a possible user input. I'd say to add one to category too then. But maybe you could explain in the JavaDoc why you decided to test this specific case explicitly?

Reply:
Makes sense. And this reminded me of another potential problem: 

> If the user successfully edits a Category from name A to B, and then edits a 

> expense that has Category type A while not changing name, value, and Category. 

> Then it will not b successful because the Category now has name B. 

@katarinaavucic @luluwuwu 
I'm wondering if we have anything that prevents this, if not, it should probably by implemented here too. (One solution I could think of is to create a new Category with name B, and edit all expenses with Category A to having Category B. Then delete Category with name A.)

Reply:
This should not cause a problem: A Category Object rather than just the category name is passed in as the ExpenseID. So when a user edits a category name that had expenses belonging to this category,  any expense.getCategory() should return the Category object with the new name!

Reply:
Right! Thanks! I got totally confused... Then I believe there's no more problems!

Reply:
Perfect, I will implement one in Category and add info to the Javadoc then!